autonomic-administration-plugin/src/main/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgent.java

```diff
@@ -176,22 +183,24 @@ private ClusterVO getClusterToManageFromZone(ClusterAdministrationHeuristicAlgor
     }
 
     /**
-     * Applies a given administration algorithm onto a cluster. First it
-     * executes the {@link #processCluster(ClusterVO, ClusterAdministrationHeuristicAlgorithm)} method;
-     * then, it sets the cluster 'administration_status' to {@link ClusterAdministrationStatus#Done} and
+     * It applies a given administration algorithm onto a cluster. First it
+     * executes the {@link #processCluster(ClusterVO, ClusterAdministrationHeuristicAlgorithm)}
+     * method;
+     * then, it sets the cluster 'administration_status' to {@link ClusterAdministrationStatus#Done}
+     * and
      * 'last_administration' to the current date at the 'cloud.cluster' table.
      */
-    private void workOnCluster(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
+    public void workOnCluster(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
         long clusterId = cluster.getId();
-        logger.debug(String.format("Managing cluster[%d].", clusterId));
+        logger.debug(String.format("Managing cluster [id=%d].", clusterId));
```

Comment:
Why not mark the AdministrationStatus inside this method instead of getClusterToManageFromZone.

Also, this method is public, can it be called on a cluster that is currently being worked on? Is this intended?